phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/MultithreadedRunner.java

```diff
@@ -0,0 +1,171 @@
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.apache.phoenix.pherf.workload;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.Calendar;
+import java.util.Date;
+
+import org.apache.phoenix.pherf.PherfConstants.RunMode;
+
+import org.apache.phoenix.pherf.result.DataModelResult;
+import org.apache.phoenix.pherf.result.ResultManager;
+import org.apache.phoenix.pherf.result.RunTime;
+import org.apache.phoenix.pherf.result.ThreadTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.phoenix.pherf.configuration.Query;
+import org.apache.phoenix.pherf.util.PhoenixUtil;
+
+class MultithreadedRunner implements Runnable {
+	private static final Logger logger = LoggerFactory
+			.getLogger(MultithreadedRunner.class);
+	private Thread t;
+	private Query query;
+	private ThreadTime threadTime;
+	private PhoenixUtil pUtil = new PhoenixUtil();
+	private String threadName;
+	private DataModelResult dataModelResult;
+	private long numberOfExecutions;
+	private long executionDurationInMs;
+	private static long lastResultWritten = System.currentTimeMillis() - 1000;
+    private final ResultManager resultManager;
+
+    /**
+     * Multithreaded runner
+     *
+     * @param threadName
+     * @param query
+     * @param dataModelResult
+     * @param threadTime
+     * @param numberOfExecutions
+     * @param executionDurationInMs
+     */
+    MultithreadedRunner(String threadName,
+                        Query query,
+                        DataModelResult dataModelResult,
+                        ThreadTime threadTime,
+                        long numberOfExecutions,
+                        long executionDurationInMs) {
+        this.query = query;
+        this.threadName = threadName;
+        this.threadTime = threadTime;
+        this.dataModelResult = dataModelResult;
+        this.numberOfExecutions = numberOfExecutions;
+        this.executionDurationInMs = executionDurationInMs;
+        this.resultManager = new ResultManager(dataModelResult.getName(), RunMode.PERFORMANCE);
+    }
+
+	/**
+	 * Executes run for a minimum of number of execution or execution duration
+	 */
+	public void run() {
+		logger.info("\n\nThread Starting " + t.getName() + " ; " + query.getStatement() + " for "
+				+ numberOfExecutions + "times\n\n");
+		Long start = System.currentTimeMillis();
+		for (long i = numberOfExecutions; (i > 0 && ((System
+				.currentTimeMillis() - start) < executionDurationInMs)); i--) {
+			try {
+                synchronized (resultManager) {
+                    timedQuery();
+                    if ((System.currentTimeMillis() - lastResultWritten) > 1000) {
+//					resultUtil.writeResultToFile(dataModelResult, dataModelResult.getName(), RunMode.PERFORMANCE);
```

Comment:
Removed. The write method takes care of it now. That was from old implementation.